eureka2-client/src/main/java/com/netflix/eureka2/client/EurekaInterestClientBuilder.java

```diff
@@ -0,0 +1,128 @@
+        this.readServerResolver = readServerResolver;
+        return self();
+    }
+
+    /**
+     * Connect to a specific read server specified by the given hostname and registrationPort
+     *
+     * @param hostname the hostname for the read server to connect to.
+     * @param interestPort the registration port for the read server
+     * @return a builder to continue client construction
+     */
+    public EurekaInterestClientBuilder fromHostname(String hostname, int interestPort) {
+        this.readServerResolver = ServerResolvers.just(hostname, interestPort);
+        return self();
+    }
+
+    /**
+     * Resolve target read servers via the dnsName provided, and then connect to the resolved read servers
+     * on the provided interestPort.
+     *
+     * @param dnsName a dns name from which to resolve read server hostnames for connection
+     * @param interestPort the registration port for the read servers
+     * @return a builder to continue client construction
+     */
+    public EurekaInterestClientBuilder fromDns(String dnsName, int interestPort) {
+        this.readServerResolver = ServerResolvers.forDnsName(dnsName, interestPort);
+        return self();
+    }
+
+    /**
+     * Resolve target read servers by first connecting to a write server with the interest protocol to pre-discover
+     * available read servers hostnames and supported interestPorts. The initial write server for pre-discovery is
+     * resolved from dnsName.
+     *
+     * @param writeDnsName a dns name from which to resolve write server hostnames for connection
+     * @param writeInterestPort the interest port for the *write* servers to pre-discover read servers
+     * @param readServerVip vip address for read server identification from the write server registry
+     * @return a builder to continue client construction
+     */
+    public EurekaInterestClientBuilder fromWriteInterestResolver(ServerResolver writeInterestResolver, String readServerVip) {
+        this.readServerResolver = ServerResolvers.fromWriteServer(writeInterestResolver, readServerVip);
+        return self();
+    }
+
+    /**
+     * Resolve target read servers by first connecting to a write server with the interest protocol to pre-discover
+     * available read servers hostnames and supported interestPorts. The initial write server for pre-discovery is
+     * resolved from dnsName.
+     *
+     * @param writeDnsName a dns name from which to resolve write server hostnames for connection
+     * @param writeInterestPort the interest port for the *write* servers to pre-discover read servers
+     * @param readServerVip vip address for read server identification from the write server registry
+     * @return a builder to continue client construction
+     */
+    public EurekaInterestClientBuilder fromWriteDns(String writeDnsName, int writeInterestPort, String readServerVip) {
```

Comment:
I would keep single ```fromInterest(ServerResolver, InstanceInfoMatcher)```. A user can create desired ServerResolver type from ```ServerResolvers```. Otherwise we will have to sync both classes, and add new resolver types in multiple places.